doc/reference/javascript_reference.rst

```diff
@@ -803,8 +814,31 @@ Widget Guidelines
 
 * All interactive components (components displaying information to the screen
   or intercepting DOM events) must inherit from :class:`~Widget`
-  and correctly implement and use its API and life cycle.
+  and correctly implement and use its API and life cycle.+
+
+* Make sure to wait for start to be finished before using $el e.g.:
+
+    .. code-block:: javascript
 
+        var Widget = require('web.Widget');
+
+        var IncorrectWidget = Widget.extend({
+            start: function () {
+                this._super.apply(arguments);
+                this.$el.hasClass(....) // First mistake: might fail because super in not finished
```

Comment:
actually, the $el is created between willStart and start, so you can use it even before calling _super